code/go/0chain.net/smartcontract/stakepool/edb_delegate_pool.go

```diff
@@ -34,31 +32,33 @@ func (dp DelegatePool) emitNew(
 
 		Status:       int(dp.Status),
 		RoundCreated: balances.GetBlock().Round,
-	})
-	if err != nil {
-		return err
 	}
+
 	balances.EmitEvent(
 		event.TypeStats,
 		event.TagAddOrOverwriteDelegatePool,
 		providerId,
-		string(data),
+		data,
 	)
 	return nil
 }
 
 func (dpu DelegatePoolUpdate) emitUpdate(
 	balances cstate.StateContextI,
 ) error {
-	data, err := json.Marshal(&dpu)
-	if err != nil {
-		return err
-	}
+
 	balances.EmitEvent(
 		event.TypeStats,
 		event.TagUpdateDelegatePool,
 		dpu.PoolId,
-		string(data),
+		delegatePoolUpdateToDbsDelegatePoolUpdate(dpu),
 	)
 	return nil
 }
+
+func delegatePoolUpdateToDbsDelegatePoolUpdate(dpu DelegatePoolUpdate) dbs.DelegatePoolUpdate {
```

Comment:
This function name is too long and repeating, perhaps use `newDBDelegatePoolUpdate()`

Reply:
fixed